portfolio/src/main/java/com/google/sps/data/BookReader.java

```diff
@@ -18,17 +19,30 @@
   public BookReader(String path) {
     this.file = new File(path);
   }
-
+ 
   public ImmutableMap<Integer, Book> makeBookList() throws IOException {
+          System.out.println("makeBookList");
     ImmutableMap.Builder<Integer, Book> allBooks = new Builder<Integer, Book>();
     try (Scanner scanner = new Scanner(file, "utf-8").useDelimiter("\\Z")) {
       String content = scanner.next().replaceAll("[\\r\\n]+", "");
       String[] lines = content.split("NEXTBOOK"); // lines[i] represents one row of the file
-
+ 
+          System.out.println("Lines: "+lines);
       String current_title = "";
-      Book.Builder current_builder = Book.builder().title("null");
+      Book.Builder current_builder = Book.builder().title("null")
```

Comment:
Needed to add these empty field or else the compiler would give an error that the builder was missing required fields.

Reply:
You could move the default values in the book class itself:

```
public static Builder builder() {
    return new AutoValue_Book.Builder().language("N/A").pageCount("N/A")...;
  }
```

Reply:
I didn't realize that. That is so much better than adding that list each time.